config/jobs/kubernetes-sigs/kueue/kueue-presubmits-main.yaml

```diff
@@ -1,3 +1,12 @@
+presets:
+- labels:
+    preset-gomaxprocs: "true"
+  env:
+  - name: GOMAXPROCS
+    valueFrom:
+      resourceFieldRef:
+        resource: limits.cpu
```

Comment:
this is very elegant, later I would like to make this a default setting for all jobs, but probably not until after 1.31 releases

Reply:
though, this will break if anyone sets non-integer values.

Reply:
no, it won't

I did a test:

```yaml
apiVersion: v1
kind: Pod
metadata:
  name: ubuntu
spec:
  restartPolicy: Never
  containers:
  - name: millicore
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 1
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "250m"
  - name: empty
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # phisical core number on the host, I got 16
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
  - name: millicore2048
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 3
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: "2048m"
  - name: point5
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 1
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: 0.5
  - name: 3point5
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # round up to 4
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: 3.5
  - name: two
    image: ubuntu:latest
    command: [/bin/bash, -c, echo $GOMAXPROCS] # print 2
    env:
    - name: GOMAXPROCS
      valueFrom:
        resourceFieldRef:
          resource: limits.cpu
    resources:
      requests:
        memory: "64Mi"
        cpu: "250m"
      limits:
        memory: "128Mi"
        cpu: 2
```